scripts/js/src/runtimeUpgrade.ts

```diff
@@ -84,18 +83,32 @@ async function doRuntimeUpgrade(
             return byteArray.reduce((str, byte) => str + byte.toString(16).padStart(2, '0'), '0x');
         };
 
-        // submit the upgrade transaction
-        const unsub = await api.tx.sudo
-            .sudoUncheckedWeight(api.tx.system.setCode(u8aToHex(wasmBlob)), 1)
-            .signAndSend(keyring, { nonce: -1 }, (result) => {
-                if (result.isInBlock && !result.isError) {
-                    console.log('Runtime upgrade successful');
-                    unsub();
-                } else if (result.isError) {
-                    console.error(`Runtime upgrade failed: ${result.toString()}`);
-                    unsub();
-                }
-            });
+        const scheduleDelay = 50;
+
+        const hexBlob = u8aToHex(wasmBlob);
+        // schedule the upgrade
+        await new Promise<void>((resolve, reject) => {
+            const unsubscribe = api.tx.sudo
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                .sudo(api.tx.scheduler.scheduleAfter(scheduleDelay, null, 0, { Value: api.tx.system.setCode(hexBlob) }))
+                .signAndSend(keyring, { nonce: -1 }, (result) => {
+                    const finish = () => {
+                        unsubscribe
+                            .then((unsub) => {
+                                unsub();
+                                resolve();
+                            })
+                            .catch(reject);
+                    };
+                    if (result.isInBlock && !result.isError) {
+                        console.log('Runtime upgrade successfully scheduled');
+                        finish();
+                    } else if (result.isError) {
+                        console.error(`Failed to schedule runtime upgrade: ${result.toString()}`);
+                        finish();
+                    }
```

Comment:
Won't this try to resolve while executing `finish()`?

Reply:
Yeah, it's probably better to reject in the error case, then catch the error and exit with a failure code at the top level. I pushed a new commit to do that.

(Is that what you meant, or did I misunderstand your comment?)

Reply:
Yup, exactly.